src/main/java/duke/component/Parser.java

```diff
@@ -0,0 +1,88 @@
+package duke.component;
+
+import duke.command.*;
+import duke.task.*;
+
+/**
+ * Interpreter of user's input
+ */
+public class Parser {
+
+    /**
+     * Parses an input string and gives the corresponding command
+     * @param input the input typed in by the user
+     * @return the command that corresponds to the input
+     * @throws DukeException if the input is invalid or illegal
+     */
+    public static Command parse(String input) throws DukeException {
+
+        if (input.equals("bye")) {
+            return new ByeCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("find")) {
```

Comment:
I like the naming of your method "startsWith" for input as it sounds very intuitive.